From e07ebc04d9a66f685c1d692a7fa825b669f49db2 Mon Sep 17 00:00:00 2001
From: Gwenael Treuveur <gwenael.treuveur@foss.st.com>
Date: Fri, 22 Nov 2024 17:16:45 +0100
Subject: [PATCH 1386/1695] dts: stm32: add OEM rproc public key hash for
 stm32mp25

Add new OTP entry for remoteproc authentication key hash.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git b7d699bedbf59448eac9a6fcd28f10f7ffce2504 dts: stm32: add OEM rproc public key hash for stm32mp25]
Change-Id: I8e163e319750ef3964a1b7c2a42e3c3731d8f8fd
Signed-off-by: Gwenael Treuveur <gwenael.treuveur@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/420442
Tested-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/dts/stm32mp251.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index eea0d23b6..0c06413c5 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -2918,6 +2918,12 @@
 				reg = <0x240 0x20>;
 			};
 
+			oem_rproc_pkh: oem_rproc_pkh@2c0 {
+				/* 0x2C0 -> otp176 */
+				reg = <0x2C0 0x20>;
+				st,non-secure-otp-provisioning;
+			};
+
 			oem_fip_enc_key: otp260@410 {
 				reg = <0x410 0x20>;
 				st,non-secure-otp-provisioning;
-- 
2.47.2

